tests/RobotFramework/tests/cumulocity/benchmarks/benchmark.py

```diff
@@ -0,0 +1,503 @@
+        LOG.info("Waiting for last message to be published")
+        self.wait_until_finished(2.0)
+
+        LOG.info("Stopping mqtt client")
+        client.disconnect()
+        client.loop_stop()
+
+        total_cloud_messages = len(
+            {
+                msg["msgid"]["msgid"]["value"]: msg
+                for msg in self.__c8y_messages
+                if msg.get("msgid", {}).get("msgid", {}).get("value", None) is not None
+            }
+        )
+        dropped_messages = self.count - total_cloud_messages
+
+        return {
+            "worker": procID,
+            "messages": self.count,
+            "dropped_percent": round(dropped_messages * 100 / self.count, 2),
+            "dropped_messages": dropped_messages,
+            "total": delta.total_seconds(),
+            "total_non_idle": round(delta.total_seconds() - idle_sec, 5),
+            "idle": idle_sec,
+            "parameters": {
+                "count": self.count,
+                "beats": self.beats,
+                "beats_delay": self.beats_delay,
+                "period": self.period,
+            },
+            "qos": self.qos,
+            "total_payload_bytes": payload_bytes,
+            "total_payload_bytes_per_message": payload_bytes / self.count,
+            "messages_per_second": round(self.count / delta.total_seconds(), 0),
+            "ms_per_message": round((delta.total_seconds()) * 1000 / self.count, 3),
+        }
+
+
+def do_work(opts):
+    return Pub(
+        host=opts.host,
+        port=opts.port,
+        topic_root=opts.topic_root,
+        topic_id=opts.topic_id,
+        topic=opts.topic,
+        count=opts.count[0],
+        qos=opts.qos,
+        beats=opts.beats[0],
+        beats_delay=opts.beats_delay[0],
+        period=opts.period[0],
+        timeout=opts.pub_timeout,
+        max_datapoints=opts.max_datapoints,
+    ).run
+
+
+def common_cli(parser: argparse.ArgumentParser):
+    parser.add_argument("--host", default="127.0.0.1")
+    parser.add_argument("--port", type=int, default=1883)
+    parser.add_argument(
+        "--iterations",
+        type=int,
+        default=1,
+        help="Number of times to run the benchmarks",
+    )
+    parser.add_argument(
+        "--mqtt-device-topic-id", dest="topic_id", default="device/main//"
+    )
+    parser.add_argument("--mqtt-topic-root", dest="topic_root", default="te")
+    parser.add_argument("--topic", default="m/environment")
+    parser.add_argument("--pub-timeout", dest="pub_timeout", default=60, type=float)
+    parser.add_argument("--pub-clients", dest="pub_clients", default=1, type=int)
+
+    parser.add_argument("--qos", default=0, type=int)
+    parser.add_argument("--max-datapoints", dest="max_datapoints", default=1, type=int)
+
+    parser.add_argument(
+        "--count",
+        default=[50],
+        type=flag_range,
+        help="Number of messages to send in one iteration",
```

Comment:
```suggestion
        help="Total number of messages to send in one iteration as multiple bursts",
```

Reply:
Yes, the api might need some cleaning up, however it is still only an internal benchmarking script so that we have coverage...though the tests seem to be a bit flakey as I don't think we can guarantee the performance on the CI runner.

Reply:
I've cleaned up the script now. Any further improvements to the script can be covered by another PR.